### PR TITLE
Dynamic roles and permissions: remove old code

### PR DESCRIPTION
**Pull request by luontola**
Cleanup of #869

### CODE REVIEW

File: src/clj/rems/workflow/dynamic.clj [hunk 1]
```diff
@@ -615,104 +544,66 @@
...
-                :event/time (:time cmd)
-                :event/actor (:actor cmd)
-                :application/member (:member cmd)
-                :application/id (:application-id cmd)
-                :application/comment (:comment cmd)}}))
+      (ok {:event/type :application.event/member-uninvited
+           :application/member (:member cmd)
+           :application/comment (:comment cmd)})))
+
+(defn- enrich-event [event cmd]
```
Comment (Macroz) [inline src/clj/rems/workflow/dynamic.clj]: I think it's better to name this more explicitly like `command->event-base` or `add-common-event-fields-from-command` style.

File: src/cljs/rems/roles.cljs [hunk 2]
```diff
@@ -22,4 +22,5 @@
              roles))
 
 (defn is-admin? [roles]
+  ;; TODO: rename owner role to admin role?
```
[the remark this answers is not in the document]
Comment (luontola) [reply src/cljs/rems/roles.cljs]: So later we'll also have an admin role, right? Do the permissions of an owner and an admin differ? Is a domain owner similar to an owner, except that he has permissions only for resources with a particular organization? I suppose there is no support for domain owners yet, except for the organization field in resources, forms and workflows.

File: src/clj/rems/permissions.clj [hunk 3]
```diff
@@ -63,6 +63,20 @@
       (is (= #{:role-2} (user-roles app "user-2")))
       (is (= #{} (user-roles app "user-3"))))))
 
+(defn has-any-role? [application user]
+  (not (empty? (user-roles application user))))
```
[the remark this answers is not in the document]
Comment (luontola) [reply src/clj/rems/permissions.clj]: I know the idiom and I disagree with it. I think "not empty?" describes better the intent of checking whether a collection is non-empty. It's a Clojure convention that methods ending in a question mark should return a boolean, so I prefer the principle of least surprise and also [robustness principle/Postel's law](https://en.wikipedia.org/wiki/Robustness_principle)

File: src/clj/rems/permissions.clj [hunk 4]
```diff
@@ -63,6 +63,20 @@
...
+
+(deftest test-has-any-role?
+  (testing "no roles"
+    (is (false?
```
[the remark this answers is not in the document]
Comment (luontola) [reply src/clj/rems/permissions.clj]: These are the tests for that function. Per the robustness principle, it is strict in what it returns - [predicates should end in a question mark and return a boolean](https://github.com/bbatsov/clojure-style-guide#pred-with-question-mark). The places where that method is used, on the other hand, should be liberal in what they accept - they should not be using `true?` or `false?`.

File: src/clj/rems/permissions.clj [hunk 5]
```diff
@@ -63,6 +63,20 @@
...
+         (-> {}
+             (has-any-role? "user")))))
+  (testing "some roles"
+    (is (true?
```
[the remark this answers is not in the document]
Comment (luontola) [reply src/clj/rems/permissions.clj]: Same as above.

File: src/cljs/rems/roles.cljs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Macroz) [reply src/cljs/rems/roles.cljs]: Domain owner has basically all rights for his/her domain, but admin only has the permission to create domain owners. Maybe some reports etc. will also be different. But yes there is no support for a domain owner yet, only a kind of super owner.

File: src/clj/rems/permissions.clj (hunk 3 above)
[the remark this answers is not in the document]
Comment (Macroz) [reply src/clj/rems/permissions.clj]: In Clojure do as the Clojurians do, hence `seq` is best.

File: src/clj/rems/permissions.clj (hunk 4 above)
[the remark this answers is not in the document]
Comment (Macroz) [reply src/clj/rems/permissions.clj]: It should not test that it's a boolean if there is no reason. In Clojure the practice is to use any non-nil and false to mean true.

File: src/clj/rems/permissions.clj (hunk 5 above)
[the remark this answers is not in the document]
Comment (Macroz) [reply src/clj/rems/permissions.clj]: Yes.

File: src/clj/rems/permissions.clj (hunk 3 above)
[the remark this answers is not in the document]
Comment (luontola) [reply src/clj/rems/permissions.clj]: done

**Pull request merged**